lib/TokenManager.js

```diff
@@ -24,64 +24,7 @@ function emitExpired(tokenMgmtRef, key, token) {
   tokenMgmtRef.emitter.emit('expired', key, token);
 }
 
-function clearRefreshTimeout(tokenMgmtRef, key) {
-  clearTimeout(tokenMgmtRef.refreshTimeouts[key]);
-  delete tokenMgmtRef.refreshTimeouts[key];
-}
-
-function clearRefreshTimeoutAll(tokenMgmtRef) {
-  var refreshTimeouts = tokenMgmtRef.refreshTimeouts;
-  for(var key in refreshTimeouts) {
-    if (!refreshTimeouts.hasOwnProperty(key)) {
-      continue;
-    }
-    clearRefreshTimeout(tokenMgmtRef, key);
-  }
-  tokenMgmtRef.refreshTimeouts = {};
-}
-
-function setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token) {
-  var refreshWait = (token.expiresAt * 1000) - Date.now();
-  if (refreshWait < 0) {
-    // Already expired
-    refreshWait = 0;
-  }
-  var refreshTimeout = setTimeout(function() {
-    if (tokenMgmtRef.autoRefresh) {
-      return refresh(sdk, tokenMgmtRef, storage, key);
-    } else if (token.expiresAt * 1000 <= Date.now()) {
-      remove(tokenMgmtRef, storage, key);
-      emitExpired(tokenMgmtRef, key, token);
```

Comment:
If I want to have more manual control over token refreshing, this timeout event is still useful (authJS can tell me when the token has expired, rather than having to setup this timeout myself).

Reply:
I want to make sure I understand what this change is doing: `setRefreshTimeout` will no longer automatically refresh the token, but will still emit an `expired` event. Is that right?

Reply:
We're doing two things inside of this timeout:
1. Emitting an `expired` event
    - This is **new**. We used to only emit this when the Okta session was dead and tokens could no longer be refreshed, or `autoRefresh` is disabled and the token expires.
2. Calling our `getAsync` method, where the token will be refreshed **if** `autoRefresh` is enabled and the token is expired.

The only change in behavior here is we're emitting an `expired` event **everytime** the token expires.